src/main/groovy/com/lesfurets/jenkins/unit/PipelineTestHelper.groovy

```diff
@@ -454,7 +454,16 @@ class PipelineTestHelper {
                 SCRIPT_SET_BINDING.invoke(script, binding)
                 script.metaClass.getMethods().findAll { it.name == 'call' }.forEach { m ->
                     this.registerAllowedMethod(method(e.value.class.name, m.getNativeParameterTypes()),
-                                    { args -> m.doMethodInvoke(e.value, args) })
+                        { args ->
+                            // When calling a one argument method with a null argument the
+                            // Groovy doMethodInvoke appears to incorrectly assume a zero
+                            // argument call signature for the method yielding an IllegalArgumentException
+                            if (args == null && m.getNativeParameterTypes().size() == 1) {
+                                m.doMethodInvoke(e.value, MetaClassHelper.ARRAY_WITH_NULL)
+                            } else {
+                                m.doMethodInvoke(e.value, args)
+                            }
+                        })
```

Comment:
```suggestion
                        { args -> m.doMethodInvoke(e.value, args ?: MetaClassHelper.ARRAY_WITH_NULL) })
```

The Elvis operator is a bit more succinct here.

Reply:
I do not believe it is for all use cases. The additional check for number of parameters is needed. args may be null for a zero argument method call.

Reply:
Ah, I see. Shouldn't it be `m.getNativeParameterTypes().size() > 1` then?

Reply:
In my opinion this code in Groovy is incorrectly handling the `null` case
[invoked Groovy](https://github.com/apache/groovy/blob/master/src/main/java/org/codehaus/groovy/reflection/ParameterTypes.java#L161)

The line in question should assign EMPTY_ARRAY to the null argumentArray and allow the subsequent lines decide if ARRAY_WITH_NULL is instead appropriate.

Reply:
No, more than one null arguments would have manifested as array of null. I can add additional tests for the behavior with multiple arguments but I believe the null single argument is an edge case not handled properly.

Reply:
Ok, I'm a bit unfamiliar with the background of the original failure here. I need to sit down and read through the linked issue before approving this PR. However, if another maintainer sees that everything is alright, I'm fine with them merging this (hence my leaving neutral feedback and not "requested changes").